### ISSUE TITLE
Different inference results from python and C# 

### ISSUE THREAD

**Issue by yangmingwanli** [microsoft/onnxruntime#10863]
**Describe the bug**
Getting different ONNX runtime inference results from the same model (resnet50 for feature extraction) in python and C#.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- ONNX Runtime installed from (source or binary):
- ONNX Runtime version:
- Python version:
- Visual Studio version (if applicable):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**To Reproduce**
**Model export to ONNX and ONNX runtime inference in Python**

from tensorflow.keras.applications.resnet50 import ResNet50
from tensorflow.keras.preprocessing import image
from tensorflow.keras.applications.resnet50 import preprocess_input
import numpy as np
import tf2onnx
import onnxruntime as rt

### pre-process image
img_path = 'ade20k.jpg'
img = image.load_img(img_path, target_size=(224, 224))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

### load model and run a test inference
model = ResNet50(include_top = False, weights = 'imagenet', pooling = 'avg' )
preds = model.predict(x)
print(preds)
>array([[0.33036256, 0.01547315, 0.05512593, ..., 0.409093  , 0.02243927,
        0.6776489 ]], dtype=float32)

### export to onnx model
spec = (tf.TensorSpec((None, 224, 224, 3), tf.float32, name="input"),)
output_path = model.name + ".onnx"

model_proto, _ = tf2onnx.convert.from_keras(model, input_signature=spec, opset=13, output_path=output_path)
output_names = [n.name for n in model_proto.graph.output]

### test inference with exported onnx model with onnx runtime in python
output_path = model.name + ".onnx"
providers = ['CPUExecutionProvider']
m = rt.InferenceSession(output_path, providers=providers)
pred_onnx = m.run(output_names, {"input": x})
print(pred_onnx)
>[array([[0.3303622 , 0.01547297, 0.05512602, ..., 0.40909338, 0.02243922,
        0.67764884]], dtype=float32)]

**The same model exported above ONNX runtime inference in C#**
Following the tutorial https://onnxruntime.ai/docs/tutorials/resnet50_csharp.html, I only modified image preprocess part to tensorflow default preprocess_input mode: caffe (convert the images from RGB to BGR,  then zero-center each color channel with respect to the ImageNet dataset, without scaling.) https://github.com/keras-team/keras/blob/master/keras/applications/resnet.py#L504-L508

            // Preprocess image
            Tensor<float> inputimg = new DenseTensor<float>(new[] { 1, 224, 224, 3 });

            for (int y = 0; y < image.Height; y++)
            {
                Span<Rgb24> pixelSpan = image.GetPixelRowSpan(y);
                for (int x = 0; x < image.Width; x++)
                {
                    inputimg[0, y, x, 0] = pixelSpan[x].B - 103.939f;
                    inputimg[0, y, x, 1] = pixelSpan[x].G - 116.779f;
                    inputimg[0, y, x, 2] = pixelSpan[x].R - 123.68f;
                }
            }
                           
                         …

            // Run inference
            using var session = new InferenceSession(modelFilePath);
            using IDisposableReadOnlyCollection<DisposableNamedOnnxValue> results = session.Run(inputs);
            Console.WriteLine("Resnet50 embedding...");
            foreach (var t in results.First().AsEnumerable<float>())
            {
                Console.WriteLine(t);
            }

C# inference

C:\Users\xxxxx\onnxruntime\csharp\sample\Microsoft.ML.OnnxRuntime.ResNet50v2Sample>dotnet run D:\onnx\resnet.onnx D:\onnx\ade20k.jpg
Resnet50 embedding...
0.2459586
0.0011222125
0.0069955415
…
0.31308472
0.08043517
0.19800954

**Expected behavior**
 onnx runtime inference results from the same model are very close in Python and C#

**Comment by hariharans29** [comment]
"I only modified image preprocess part..." - Such cross-language discrepancies have been mostly attributed to this tiny detail. Can you please check if the raw input tensors being fed in are the same in Python and C# (after the pre-processing step)?

**Comment by wschin** [comment]
Agree with @hariharans29. Floating-point operations are implementation-dependent. To reduce investigation scope, we need to make sure the input to ORT is the same.
